### PR TITLE
Remove create-stable-version action and use regexes

### PR DESCRIPTION
**Pull request by MitchLillie**
### Background

This automation didn't quite work due to missing perms, and it was a bit clunky anyways as the PR this attempted to create needs to be merged _before_ the action runs.

### Solution

(Describe your solution, why this approach was chosen, and what the alternatives/impacts may be)

### 🎩

- ...

### Checklist

- [ ] I have :tophat:'d these changes
- [ ] I have updated relevant documentation

### CODE REVIEW

File: .github/workflows/create-stable-version.yml [hunk 1]
```diff
@@ -33,16 +33,6 @@ jobs:
...
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v5
-        with:
-          token: ${{ secrets.SHOPIFY_GH_ACCESS_TOKEN }}
-          commit-message: 'Add ${{ github.event.inputs.version }} to watched branches'
-          title: 'Add ${{ github.event.inputs.version }} to watched branches'
-          body: 'This PR adds the new stable version ${{ github.event.inputs.version }} to the list of watched branches in the changesets-reminder and deploy GitHub actions.'
-          branch: 'update-watched-branches-${{ github.event.inputs.version }}'
-          base: '${{ github.event.inputs.version }}'
-
```
Comment (jamesvidler) [inline .github/workflows/create-stable-version.yml:45]: You are removing the PR that gets created, but you still create the branch with the updated GH action files? I'm not sure if I understand correctly.

Comment (MitchLillie) [reply .github/workflows/create-stable-version.yml:45]: Ahh I see what you mean. Yes good call, we don't need that either.

**Review by jamesvidler — COMMENTED** [review]
I see a slight problem with this existing workflow. We update the "latest version" variable a little too early in the process. Sometimes a stable version might be cut for days before it actually ships. 

Consider that we create a branch for 2025-01 and the "latest version" is updated to that. But, while it is in review a patch was shipped for 2024-10. 2024-10 is still technically the latest version, but it won't get tagged as "latest" because the "latest version" variable has already been updated to `2025-01`.

Maybe we can find another way to identify the "latest" version?

**Comment by MitchLillie** [comment]
@jamesvidler Good point. If we remove that part, all the action did was create a new branch which isn't very helpful. I updated the PR to remove the new action entirely and use the new regexes where appropriate.

**Pull request merged**